cmd/cloud.go

```diff
@@ -148,11 +138,15 @@ func (c *cmdCloud) run(cmd *cobra.Command, args []string) error {
 	if arc.Options.External == nil {
 		arc.Options.External = make(map[string]json.RawMessage)
 	}
-	arc.Options.External["loadimpact"], err = json.Marshal(tmpCloudConfig)
+
+	b, err := json.Marshal(tmpCloudConfig)
 	if err != nil {
 		return err
 	}
 
+	arc.Options.Cloud = b
+	arc.Options.External[cloudapi.LegacyCloudConfigKey] = b
```

Comment:
```suggestion
	arc.Options.SetCloud(b)
```
Can we add a method directly a method on the Options type? In this way, we guarantee the _atomicity_ of the operation in the future preventing bugs, in case we change the code here.

Reply:
On the one side, I'm with you. On the other side, I don't think that this should be publically exposed with the API :thinking:

Reply:
Also important to say that this is more for safety, and the future version should have only `arc.Options.Cloud = b`

Reply:
>On the other side, I don't think that this should be publically exposed with the API 🤔

I'm not getting it. `arc.Options.Cloud = b` is going to be exported, so what will be the difference?

>Also important to say that this is more for safety, and the future version should have only arc.Options.Cloud = b

I got the feeling we would like to keep the 2-way forever. Have we changed the plan around it?

Reply:
> I'm not getting it. arc.Options.Cloud = b is going to be exported, so what will be the difference?

The difference is that the `Cloud` and `Ext` are the fields that will stay almost forever, but this two-times setting is a temporary thing. So simply the method like `Options.SetCloudOptions` is also a temporary thing.

>I got the feeling we would like to keep the 2-way forever. Have we changed the plan around it?

Not, that was probably a wrong impression. We keep the support of the `options.ext.loadimpact` for a long time, but this place is writing. There is no need to continue filling out the `options.ext.loadimpact` if all backend components support the new way of `options.cloud`. So once we rich that state, there is no more need to write the ``options.ext.loadimpact` and this place will become just `arc.Options.Cloud = b`